### PR TITLE
Registration QR Code

### PR DESCRIPTION
**Pull request by nmatej1**

### CODE REVIEW

**Comment by boris-barath** [comment]
Why are we rendering the QR-reader in 2 levels deep sub components?

File: health-claim/src/components/Login/RegisterForm.js [hunk 1]
```diff
@@ -17,10 +21,23 @@ export default class RegisterForm extends Component {
...
   }
 
+  handleClick(compName, e) {
+    console.log(compName);
+    this.setState({ render: compName });
+  }
+
+  _renderSubComp() {
+    switch (this.state.render) {
+      case "QRReader":
```
Comment (boris-barath) [inline health-claim/src/components/Login/RegisterForm.js]: Why do we keep this as a string and not as a bool?

File: health-claim/src/components/QRReader/QRReaderButton.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      console.log("Starting");
+    }
+  }
+
+  _renderSubComp() {
+    switch (this.state.render) {
+      case "QRReader":
+        return <QRReader />;
+      case "NOQRReader":
+        return <></>;
```
Comment (boris-barath) [inline health-claim/src/components/QRReader/QRReaderButton.js]: Similar, why a separate component and not a boolean?

File: health-claim/src/components/Login/RegisterForm.js (hunk 1 above)
Comment (nmatej1) [reply health-claim/src/components/Login/RegisterForm.js]: Ok, good point. I simplified

File: health-claim/src/components/Login/RegisterForm.js [hunk 3]
```diff
@@ -29,42 +31,6 @@ export default class RegisterForm extends Component {
...
-            <input
-              type="submit"
-              value="Sign up"
-              className="Button Button__Green"
-            />
-          </form>
-        </Box>
-      </>
-    );
+    return <RegistrationFormBox />;
```
Comment (boris-barath) [inline health-claim/src/components/Login/RegisterForm.js]: one last thing, do we need to extract this? If we're not passing anything it seems a bit unnecessary

**Pull request closed without merging**